backend/src/nodes/utils/auto_split.py

```diff
@@ -12,6 +12,34 @@ class Split:
     pass
 
 
+def estimate_tile_size(
+    free: int,
```

Comment:
I like NCNN's term for it: budget. That way, we could also get rid of `allowed_usage`. We simply say that we are allowed some budget and want a tile size for that.